File: Quan_Ly_Thu_Vien/src/QuyenSachKhoaHoc.java

public class QuyenSachKhoaHoc extends DauSachKhoaHoc implements DacTrungQuyenSach
{
	private int sMaQuyenSach;
	private boolean bDaMuonHayChua;
	private Date dtNgayMuon;
	private Date dtNgayPhaiTra;
	
	public int getsMaQuyenSach() 
	{
		return sMaQuyenSach;
	}
	public void setsMaQuyenSach(int sMaQuyenSach) 
	{
		this.sMaQuyenSach = sMaQuyenSach;
	}
	public boolean getbDaMuonHayChua() 
	{
		return bDaMuonHayChua;
	}
	public void setbDaMuonHayChua(boolean bDaMuonHayChua)
	{
		this.bDaMuonHayChua = bDaMuonHayChua;
	}
	public Date getdtNgayMuon() 
	{
		return this.dtNgayMuon;
	}
	public void setdtNgayMuon(Date dtNgayMuon) 
	{
		this.dtNgayMuon = dtNgayMuon;
	}
	public Date getdtNgayPhaiTra()
	{
		return dtNgayPhaiTra;
	}
	public void setdtNgayPhaiTra(Date dtNgayPhaiTra) 
	{
		this.dtNgayPhaiTra = dtNgayPhaiTra;
	}
	public QuyenSachKhoaHoc()
	{
		super();
		this.bDaMuonHayChua = true;
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
	}
	public QuyenSachKhoaHoc(String MaDauSach,String Tua, String NXB,String TacGia, double Gia, String ChuyenNganh,int MaQuyenSach)
	{
		super(MaDauSach, Tua, NXB, TacGia, Gia,ChuyenNganh);
		this.sMaQuyenSach = MaQuyenSach;
		this.bDaMuonHayChua = true;						
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
		TinhThanhTien();
	}
	@Override
	public void Nhap()
	{
		super.Nhap();
	}
	@Override 
	public void Xuat()
	{
		super.Xuat();
		if (this.bDaMuonHayChua == false)
		{
			System.out.println(" Da duoc muon! ");
			System.out.println(" Ngay muon : " + this.dtNgayMuon.getiNgay() + "/" + this.dtNgayMuon.getiThang() + "/" + this.dtNgayMuon.getiNam());
			System.out.println(" Ngay phai tra : " + this.dtNgayPhaiTra.getiNgay() + "/" + this.dtNgayPhaiTra.getiThang() + "/" + this.dtNgayPhaiTra.getiNam());
		}
		else
		{
			System.out.println(" Nam trong thu vien!  ");
		}
		System.out.print(" Ma quyen sach: " + this.sMaQuyenSach);
	}
	@Override 
	public void TinhThanhTien()
	{
		this.dThanhTien = this.dGia + (this.dGia * this.dThue/100);
	}
	
	public void MuonNgayHomNay(Date Today)
	{
		this.bDaMuonHayChua=false;
		this.dtNgayMuon = Today;
		this.dtNgayPhaiTra = Date.PlusDay(Today,60);
	}
	public void TraSach()
	{
		this.bDaMuonHayChua = true;
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
	}
	public void XDNgayTra()
	{
		this.dtNgayPhaiTra = Date.PlusDay(this.dtNgayMuon,60);
	}
}

File: ConsoleApp1/ConsoleApp1/QuyenSachKhoaHoc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
	class QuyenSachKhoaHoc : DauSachKhoaHoc,DacTrungQuyenSach
	{
		private int sMaQuyenSach;
		private bool bDaMuonHayChua;
		private Date dtNgayMuon;
		private Date dtNgayPhaiTra;
		public int MaQuyenSach
		{
			get { return this.sMaQuyenSach; }
		}
		public bool DaMuonHayChua
		{
			get { return this.bDaMuonHayChua; }
			set { this.bDaMuonHayChua = value; }
		}
		public Date NgayMuon
		{
			get { return this.dtNgayMuon; }
			set { this.dtNgayMuon = value; }
		}
		public Date NgayPhaiTra
		{
			get { return this.dtNgayPhaiTra; }
			set { this.dtNgayPhaiTra = value; }
		}
		public QuyenSachKhoaHoc() : base()
		{
			this.bDaMuonHayChua = true;
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
		}
		public QuyenSachKhoaHoc(string MaDauSach, string Tua, string NXB, string TacGia, double Gia, string ChuyenNganh,int MaQuyenSach) : base(MaDauSach, Tua, NXB, TacGia, Gia,ChuyenNganh)
		{
			this.sMaQuyenSach = MaQuyenSach;
			this.bDaMuonHayChua = true;						// true la chua duoc muon, con trong thu vien, false la da duoc muon roi
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
			TinhThanhTien();
		}
		public override void Nhap()
		{
			base.Nhap();
		}
		public override void Xuat()
		{
			base.Xuat();
			if (this.bDaMuonHayChua == false)
			{
				Console.WriteLine(" Da duoc muon! ");
				Console.WriteLine("Ngay het duoc muon : " + this.dtNgayMuon.Ngay + "/" + this.dtNgayMuon.Thang + "/" + this.dtNgayMuon.Nam);
				Console.WriteLine("Ngay het phai tra : " + this.dtNgayPhaiTra.Ngay + "/" + this.dtNgayPhaiTra.Thang + "/" + this.dtNgayPhaiTra.Nam);
			}
			else
			{
				Console.WriteLine(" Nam trong thu vien!  ");
			}
			Console.Write(" Ma quyen sach: " + this.sMaQuyenSach);
		}
		public override void TinhThanhTien()
		{
			this.dThanhTien = this.dGia + (this.dGia * this.dThue/100);
		}
		
		public void MuonNgayHomNay(Date Today)
		{
			this.bDaMuonHayChua = false;
			this.dtNgayMuon = Today;
			this.dtNgayPhaiTra = (Today + 60);
		}
		public void TraSach()
		{
			this.bDaMuonHayChua = true;
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
		}
		public void XDNgayTra()
		{
			this.dtNgayPhaiTra = this.dtNgayMuon + 60;
		}
	}
}
